conans/client/conf/detect.py

```diff
@@ -147,7 +155,15 @@ def _get_default_compiler(output):
         output.info("CC and CXX: %s, %s " % (cc or "None", cxx or "None"))
         command = cc or cxx
         if "gcc" in command:
-            return _gcc_compiler(output, command)
+            gcc = _gcc_compiler(output, command)
+            if platform.system() == "Darwin" and gcc is None:
+                output.warn(
+                    "%s detected as a frontend using apple-clang, skipping it to use native apple-clang" % command
+                )
+                # Fallback to use clang instead
+                command = "clang"
```

Comment:
I still think this is a weird outcome. So you defined that your CC=gcc, still conan is detecting and defaulting your profile to clang compiler? I think it is safer to just leave the compiler None for this case, so it will fail fast, instead of lots of potential mismatches between the default profile  and the declared default compiler CC.

Reply:
@memsharded 

>memsharded 5 days ago  Owner
>I am not sure we want to fallback to clang. Because if the env-var CC and CXX are set to gcc, >then, typically CMake will use them when running, and the user might not notice, because it >detected "clang" when CC=gcc. What do you think @lasote?

>  @lasote
>lasote 4 days ago  Owner
>I'm ok with this. If the user is messing with CC/CXX is his problem. And the default profile is >just a default.

>  @memsharded
> memsharded 4 days ago  Owner
> Ok, makes sense

Reply:
I talked with @memsharded and I changed my mind, it is better to no set a compiler and make more visible that we are not managing this situation with Conan. Not a big deal.

Reply:
@lasote @memsharded Ok now conan should display:

```
16:13 $ CC=gcc conan install libiconv/1.15@bincrafters/stable
Auto detecting your dev setup to initialize the default profile (/Users/pvicente/.conan/profiles/default)
CC and CXX: gcc, None
ERROR: gcc detected as a frontend using apple-clang. Compiler not supported
ERROR: Unable to find a working compiler
Default settings
	os=Macos
	os_build=Macos
	arch=x86_64
	arch_build=x86_64
	build_type=Release
*** You can change them in /Users/pvicente/.conan/profiles/default ***
*** Or override with -s compiler='other' -s ...s***
libiconv/1.15@bincrafters/stable: Not found in local cache, looking in remotes...
libiconv/1.15@bincrafters/stable: Trying with 'conan-center'...
libiconv/1.15@bincrafters/stable: Trying with 'conan-transit'...
....
```